addons/udes_stock/models/stock_quant.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+from odoo import models, _, api
+from collections import defaultdict
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+    _description = "Quant"
+
+    @api.model
+    def _get_removal_strategy_order(self, removal_strategy=None):
+        """Change fifo removal strategy"""
+        # NOTE: Preserve `NULLS FIRST` for `in_date` and use for `package_id` as well
+        if removal_strategy == "fifo":
+            return "in_date ASC NULLS FIRST, package_id ASC NULLS FIRST"
+        return super(Quant, self)._get_removal_strategy_order(removal_strategy)
```

Comment:
`Quant` should be `StockQuant`.

I think this was a long standing bug we recently sorted on 11. Luckily we had only been using first in first out!

Reply:
i remember! i came across it :D will get it fixed